### PR TITLE
MIG-677: Network requirements

### PR DESCRIPTION
**Pull request by apinnick**
https://issues.redhat.com/browse/MIG-677

Network requirements for MTC added to prerequisites. Couple small cleanup tasks.

4.5+

Previews:

https://deploy-preview-33431--osdocs.netlify.app/openshift-enterprise/latest/migrating_from_ocp_3_to_4/migrating-applications-3-4.html#migration-prerequisites_migrating-applications-3-4

https://deploy-preview-33431--osdocs.netlify.app/openshift-enterprise/latest/migration-toolkit-for-containers/migrating-applications-with-mtc.html#migration-prerequisites_migrating-applications-with-mtc

### CODE REVIEW

File: modules/migration-prerequisites.adoc [hunk 1]
```diff
@@ -6,35 +6,52 @@
...
+* The clusters must have no critical conditions.
+* The clusters must be in a `Ready` state.
+
+.Network
+
+* The clusters have unrestricted network access to each other and to the replication repository.
+* If you move the persistent volumes, the clusters must have unrestricted network access to the remote volumes.
+ifdef::migrating-applications-3-4[]
+* The {product-title} 3 cluster must have the following ports enabled:
+** `6443` (API server)
```
Comment (jmontleon) [inline modules/migration-prerequisites.adoc:40]: This is 8443 by default on OpenShift 3 by default, correct?

File: modules/migration-prerequisites.adoc [hunk 2]
```diff
@@ -6,35 +6,52 @@
...
+** `443` (routes)
+** `53` (DNS)
 endif::[]
+* The {product-title} 4 cluster must have the following ports enabled:
+** `443` (API server and routes)
```
Comment (jmontleon) [inline modules/migration-prerequisites.adoc:45]: API server is on 6643 on OpenShift 4. See https://api.jmontleonsrc.migration.redhat.com:6443. On OpenShift 4 the API port is not configurable.

**Review by jmontleon — CHANGES_REQUESTED** [review]
I think API port entries for both are incorrect.

File: modules/migration-prerequisites.adoc (hunk 1 above)
Comment (dymurray) [reply modules/migration-prerequisites.adoc:40]: Correct

File: modules/migration-prerequisites.adoc (hunk 2 above)
Comment (dymurray) [reply modules/migration-prerequisites.adoc:45]: 6443**

Comment (jmontleon) [reply modules/migration-prerequisites.adoc:45]: Thanks for catching my typo @dymurray, fixed in my comment.

File: modules/migration-prerequisites.adoc [hunk 3]
```diff
@@ -6,35 +6,51 @@
...
+* You must enable the following ports on an {product-title} 4 cluster:
+** `6443` (API server)
+** `443` (routes)
+** `53` (DNS)
+* You must enable port `443` on the replication repository if you are using TLS.
+
+.Persistent volumes (PVs)
+
+* The PVs must be valid.
+* The PVs must be bound to persistent volume claims.
```
Comment (XinRedhat) [inline modules/migration-prerequisites.adoc:51]: > The PVs must be bound to persistent volume claims.
I don't think this is required.  @sergiordlr right?

**Review by sergiordlr — APPROVED** [review]
Looks good to me, thank you very much!

File: modules/migration-prerequisites.adoc (hunk 3 above)
Comment (sergiordlr) [reply modules/migration-prerequisites.adoc:51]: As far as I know this is required.

**Comment by XinRedhat** [comment]
@apinnick LGTM

File: migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc [hunk 4]
```diff
@@ -9,12 +9,18 @@ toc::[]
...
+* link:https://docs.openshift.com/container-platform/3.11/install_config/registry/securing_and_exposing_registry.html#exposing-the-registry[Manually exposing a secure registry for {product-title} 3]
+
 include::modules/migration-updating-deprecated-internal-images.adoc[leveloffset=+2]
 include::modules/migration-creating-ca-bundle.adoc[leveloffset=+2]
 include::modules/migration-configuring-proxy-for-dvm.adoc[leveloffset=+2]
 include::modules/migration-writing-ansible-playbook-hook.adoc[leveloffset=+2]
 
-[id="migrating-applications-additional-resources_{context}"]
+[discrete]
 === Additional resources
```
Comment (mikemckiernan) [inline migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc:24]: Please keep an ID attribute.  Even if it isn't a target for a link from the TOC, these common headings can be useful targets for tech support when they provide URLs to customers.

Some guidelines on the IDs: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

File: migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc [hunk 5]
```diff
@@ -9,12 +9,18 @@ toc::[]
 You can migrate your applications by using the {mtc-full} ({mtc-short}) web console or from the command line.
 
 include::modules/migration-prerequisites.adoc[leveloffset=+1]
+
+[discrete]
+=== Additional resources
```
Comment (mikemckiernan) [inline migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc:14]: Can you add an ID here?  Please see my next comment.

Some day, we'll be able to put this link a little closer to that second bullet.  Our poor customers.

**Review by mikemckiernan — APPROVED** [review]
@apinnick, please keep the ID for the Additional resources headings. They can be helpful to support folks when they provide URLs to customers.

It seems a little wrong that there are multiple Additional resources headings in the assembly, but I get the sense that you're a bit stuck because you can't add links from within the modules. I don't have any better ideas.  Silly rules.

File: migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc (hunk 4 above)
Comment (apinnick) [reply migrating_from_ocp_3_to_4/migrating-applications-3-4.adoc:24]: I believe the ID attribute is not required for discrete headers. See `migrating_from_ocp_3_to_4/planning-migration-3-4.adoc`, which is maintained by Andrea Hoffer.

**Pull request merged**